pkg/core/resource_graph_test.go

```diff
@@ -38,6 +40,17 @@ type testResource struct {
 func (tr *testResource) KlothoConstructRef() []AnnotationKey {
 	return nil
 }
+func (tr *testResource) Create(dag *ResourceGraph, metadata map[string]any) (Resource, error) {
+	currName := ""
+	if strName, ok := metadata["name"].(string); ok {
+		currName = strName + uuid.NewString()
+	} else {
+		return tr, fmt.Errorf("Failure")
+	}
+
+	tr = &testResource{Name: currName}
```

Comment:
This should probably be `*tr = testResource` so it mutates `tr` (I believe this as written won't actually change the struct, just what the `tr` variable points to).